fluent-react/test/provider_valid.test.js

```diff
@@ -28,29 +28,37 @@ describe("LocalizationProvider - validation", () => {
 
   test("without a child", () => {
     expect(() => {
-      TestRenderer.create(<LocalizationProvider bundles={[]} />);
+      TestRenderer.create(<LocalizationProvider l10n={new ReactLocalization([])} />);
     }).toThrow(/required/);
   });
 
-  test("without bundles", () => {
+  test("without the l10n prop", () => {
     expect(() => {
       TestRenderer.create(<LocalizationProvider />);
     }).toThrow(/is required/);
   });
 
-  test("without iterable bundles", () => {
-    expect(() => {
-      TestRenderer.create(<LocalizationProvider bundles={0} />);
-    }).toThrow(/must be an iterable/);
-  });
-
   test("is memoized (no re-render) when props are the same", () => {
-    const bundles = [];
-    const spy = jest.spyOn(bundles, Symbol.iterator);
-    let renderer = TestRenderer.create(<LocalizationProvider bundles={bundles} />);
+    const l10n = new ReactLocalization([]);
+    const mockFn = jest.fn();
+
+    function MockComponent() {
+      mockFn();
+      return "A mock component";
+    }
+
+    let renderer = TestRenderer.create(
+      <LocalizationProvider l10n={l10n}>
+        <MockComponent />
+      </LocalizationProvider>
+    );
     act(() => {
-      renderer = renderer.update(<LocalizationProvider bundles={bundles} />);
+      renderer = renderer.update(
+        <LocalizationProvider l10n={l10n}>
+          <MockComponent />
+        </LocalizationProvider>
+      );
     });
-    expect(spy).toHaveBeenCalledTimes(1);
+    expect(mockFn).toHaveBeenCalledTimes(1);
```

Comment:
This test currently fails. I'm not sure how to test that the tree under the `Context.Provider` hasn't re-rendered. @Gregoor, do you have any ideas for how to test it?

Reply:
Don't have a fluent fork running atm, so this is a little speculative. Shouldn't you be able to do?

```jsx
const child = <MockComponent />;
let renderer = TestRenderer.create(
  <LocalizationProvider l10n={l10n}>
    {child}
  </LocalizationProvider>
);
act(() => {
  renderer = renderer.update(
    <LocalizationProvider l10n={l10n}>
      {child}
    </LocalizationProvider>
  );
});
expect(mockFn).toHaveBeenCalledTimes(1);
```
Which hopefully should be true and reflect the way it's being used?

Reply:
Ah yes, thank you, this works! I spent more time looking at this test that I'm willing to admit, and somehow I missed the fact that I was passing a new `<MockComponent />` in the `update` call... 🤦‍♂️

Reply:
That said, I'm not sure if this test tests what I'd like it to test :) I'll dig into this tomorrow.